drv/sprot-api/src/lib.rs

```diff
@@ -499,6 +510,9 @@ pub enum RspBody {
     Attest(Result<AttestRsp, AttestError>),
 
     Page(Result<RotPageRsp, UpdateError>),
+
+    // Added in sprot protocol version 6
+    Policy(PolicyRsp),
```

Comment:
One thing I'm not sure of: why do some messages use the `SprotError` as the return code, but others return a nested `Result<ThingyRsp, ThingyError>`?

Reply:
IIRC, for some errors, like `DumpError` in particular, we don't want to expose them up through the control plane agent. But also, it doesn't really make sense to embed unrelated errors inside `SprotError`. This was likely a mistake I made early on that we are living with. I'd stick with embedding a result for new messages. The result can also be embedded in the singleton type and parsed out later, so no need to add one here since it's not immediately needed.

Reply:
Alright, I switched to a dedicated `Result<StateRsp, StateError>`